Add plugin: LskyPro Upload

woodchen-ink opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:https://github.com/woodchen-ink/obsidian-lskypro-upload

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
Why did you not submit your changes to the original plugin?

[# Obsidian LskyPro Upload Plugin](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/README.md#L1-L2)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.


[Copyright (c) 2023 NekoTarou](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/LICENSE#L3)
Update this.

[console.log('Loaded settings:', this.settings);](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/main.ts#L44), [console.log('Token value:', this.settings.token);](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/uploader.ts#L35-L36), [console.log('Authorization header:', authHeader);](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/uploader.ts#L43), [console.log('All headers:');](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/uploader.ts#L64-L67)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[{](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/main.ts#L263-L266)
This is already the default, you can remove this code.

[containerEl.createEl("h2", { text: t("Plugin Settings") });](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/setting.ts#L47)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[new Setting(containerEl)](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/setting.ts#L195-L205)
This button serves no purpose, remove it.

woodchen-ink commented:
> Why did you not submit your changes to the original plugin?
> 
> [# Obsidian LskyPro Upload Plugin](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/README.md#L1-L2) Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.
> 
> [Copyright (c) 2023 NekoTarou](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/LICENSE#L3) Update this.
> 
> [console.log('Loaded settings:', this.settings);](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/main.ts#L44), [console.log('Token value:', this.settings.token);](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/uploader.ts#L35-L36), [console.log('Authorization header:', authHeader);](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/uploader.ts#L43), [console.log('All headers:');](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/uploader.ts#L64-L67) Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.
> 
> [{](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/main.ts#L263-L266) This is already the default, you can remove this code.
> 
> [containerEl.createEl("h2", { text: t("Plugin Settings") });](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/setting.ts#L47) [Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
> 
> [new Setting(containerEl)](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/817b7ed0de9cd2f51a924a8c8b1894e6f0a176dd/src/setting.ts#L195-L205) This button serves no purpose, remove it.

Because the original plugin has not been updated for a long time, and I have modified key features, deleted some functionalities from the original plugin, the original author is unlikely to merge it.
I have made changes to the points you mentioned.

And the lskypro open-source version has been notified to suspend updates, afterwards I will ensure compatibility with the new 2.0 enterprise version, which will have a different interface than the original plugin.

thanks

joethei commented:
[Copyright (c) 2025 WoodChen](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/7c683fad3273a5c6a5b56bf7eba38891dd22d532/LICENSE#L3)
You need to keep the original copyright here, and add yourself after that.

woodchen-ink commented:
> [Copyright (c) 2025 WoodChen](https://github.com/woodchen-ink/obsidian-lskypro-upload/blob/7c683fad3273a5c6a5b56bf7eba38891dd22d532/LICENSE#L3) You need to keep the original copyright here, and add yourself after that.

OK, I have fixed it
